### PR TITLE
MCS-774 and MCS-773 Linting

### PR DESCRIPTION
**Pull request by deanwetherby**
Added black python formatter and shellcheck script formatter. Fixed all formatting issues in scripts or added a shellcheck disable tag.

Shellcheck requires installation as in `brew install shellcheck` or `sudo apt install shellcheck` to work.

Configured pre-commit to also trim whitespace and ensure trailing empty line.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.4.0
+    hooks:
+    -   id: end-of-file-fixer
+    -   id: trailing-whitespace
+-   repo: https://github.com/psf/black
+    rev: 21.12b0
+    hooks:
+    -   id: black
+        args: [--line-length, '80']
```
Comment (deanwetherby) [inline .pre-commit-config.yaml:11]: black defaults to 88 characters per line. This is only a suggestion to the linter but some lines remain slightly over 80 but violate some other rule black deems more important.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+-   repo: https://github.com/jumanjihouse/pre-commit-hooks
+    rev: master
+    hooks:
+    -   id: shellcheck
```
Comment (deanwetherby) [inline .pre-commit-config.yaml:15]: pre-commit still requires OS-level shellcheck installation.

File: deploy_files/baseline/ray_script.sh [hunk 3]
```diff
@@ -4,9 +4,12 @@
 # LOCAL_DIR=$(dirname $0)
 # CHECK_PASSED_VAR_LOC="${LOCAL_DIR%/*}/check_passed_variables.sh"
 # source $CHECK_PASSED_VAR_LOC
+# shellcheck source=/dev/null
```
Comment (deanwetherby) [inline deploy_files/baseline/ray_script.sh:7]: shellcheck wants to verify the location of the file. by adding `source=/dev/null`, the check passes.

File: deploy_files/baseline/ray_script.sh [hunk 4]
```diff
@@ -4,9 +4,12 @@
...
 source /home/ubuntu/check_passed_variables.sh
+# shellcheck disable=SC2154
```
Comment (deanwetherby) [inline deploy_files/baseline/ray_script.sh:9]: disabling `referenced but unassigned`

File: pipeline/log_parser.py [hunk 5]
```diff
@@ -11,8 +11,8 @@
 import argparse
 import logging
 import os
-import re
 import pathlib
+import re
```
Comment (deanwetherby) [inline pipeline/log_parser.py:15]: black does isort

File: xserver/install_nvidia.sh [hunk 6]
```diff
@@ -2,13 +2,15 @@
 # nvidia_version=`cat /proc/driver/nvidia/version |grep 'NVRM version:'| grep -oE "Kernel Module\s+[0-9.]+"| awk {'print $3'}`
 
 nvidia_version=450.80.02
-nvidia_major_version=`echo $nvidia_version |sed "s/\..*//"`
+# shellcheck disable=SC2034
+# nvidia_major_version=$(echo $nvidia_version |sed "s/\..*//")
```
Comment (deanwetherby) [inline xserver/install_nvidia.sh:6]: nvidia_major_version didn't seem to be used anywhere (Ctrl+Shift+F search).

**Review by kdrumm — APPROVED** [review]

**Review by brianpippin — APPROVED** [review]

**Pull request merged**